services/worker-manager/src/providers/azure.js

```diff
@@ -451,49 +671,158 @@ class AzureProvider extends Provider {
     });
   }
 
+  /*
+   * Called after an iteration of the worker scanner
+   */
   async scanCleanup() {
     this.monitor.log.scanSeen({providerId: this.providerId, seen: this.seen});
+    await Promise.all(Object.entries(this.seen).map(async ([workerPoolId, seen]) => {
+      const workerPool = await this.WorkerPool.load({
+        workerPoolId,
+      }, true);
+
+      if (!workerPool) {
+        return; // In this case, the workertype has been deleted so we can just move on
+      }
+
+      if (this.errors[workerPoolId].length) {
+        await Promise.all(this.errors[workerPoolId].map(error => workerPool.reportError(error)));
+      }
+    }));
   }
 
-  async _removeWorker({resourceGroupName, location, vm, ip, nic, disk}) {
-    // clean up related resources, continue on errors
-    // delete VM, IP, NIC, and disk
-    const ignoreNotFound = async (promise) => {
+  /*
+   * removeResource attempts to delete a resource and verify deletion
```

Comment:
what does the return value mean?

Reply:
added a more detailed comment, returning `true` means we're positive the resource to delete is gone. if we return false we've made a request to delete that hasn't finished yet.